### PR TITLE
Remove usage of policy/v1beta1 API

### PR DESCRIPTION
**Pull request by vpnachev**
**What this PR does / why we need it**:
Remove usage of policy/v1beta1 API

With https://github.com/gardener/gardener/pull/8087 clusters with k8s version <= 1.21 are not supported, hence there is no need different API versions for PDBs to be used.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator

```

### CODE REVIEW

**Review by shafeeqes — COMMENTED** [review]
Thanks, lgtm.

File: pkg/controller/lifecycle/actuator.go [hunk 1]
```diff
@@ -482,7 +469,17 @@ func getSeedResources(lakomReplicas *int32, namespace, genericKubeconfigName, sh
...
 		lakomDeployment,
-		lakomPDB,
+		&policyv1.PodDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      constants.ExtensionServiceName,
+				Namespace: namespace,
+				Labels:    getLabels(),
+			},
+			Spec: policyv1.PodDisruptionBudgetSpec{
+				MaxUnavailable: intstr.ValueOrDefault(nil, intstr.FromInt(1)),
```
Comment (dimityrmirchev) [inline pkg/controller/lifecycle/actuator.go:479]: Nit: I would rather not wrap this in another function call. I find it easier and simpler to read

```golang
maxUnavailable = intstr.FromInt(1)
// and then 
MaxUnavailable: &maxUnavailable,
```

or directly 
```golang
 MaxUnavailable: &intstr.IntOrString{Type: intstr.Int, IntVal: 1}
```

**Review by dimityrmirchev — COMMENTED** [review]
@vpnachev 
Looks good! I have just one nit for which I leave the decision to you if you want to address this or not.

File: pkg/controller/lifecycle/actuator.go (hunk 1 above)
Comment (vpnachev) [reply pkg/controller/lifecycle/actuator.go]: Done

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**